### ISSUE TITLE
Create User and Role Management

### ISSUE THREAD

**Issue by codemvp** [codemvp/accountgo#1]

**Comment by angwenyi** [comment]
I want to help in this module. But i have issue in with mobx Version 3.15 module. Am using VS 2015.  
Is there a different version that can work.

**Comment by codemvp** [comment]
Hi, yeah. Want to give it a shot? Feel free to ask questions.
Thanks,
Marvin

**Comment by vivencio2** [comment]
Alright sure! Will ask questions in a while.

**Comment by marvinvperez** [comment]
Hi @vivencio2 , I just made a PR to your fork of the latest from master. I just want you to see some initial work with regards to modular and plugin feature. I saw you are looking at OrcharCore which is modular and extensible. I may use some help as well on making AccountGo modular and extensible. Please look at Infrastructure project and Module.Tests project

**Comment by vivencio2** [comment]
Hi Marvin,

I saw your PR and already merged it. It's good that your doing this modular approach, I have some experience on this using MEF for dll versioning and also have seen this in NopCommerce as well. I'm just ramping up my setup and will definitely help you on this.

**Comment by marvinvperez** [comment]
Awesome! We could use NopCommerce, DotNetCorePlugins and SimplCommerce as our guide.

**Comment by vivencio2** [comment]
Hi Marvin, I'm done with my setup and successfully build and run the both API and Web. So what's your plan on this user and role management? I noticed you are already started using ASPNET Identity/Membership, I can continue on that as well but I just want to know what are the plan roles you want ? Do you also want to enable social login providers ?

**Comment by codemvp** [comment]
Yeah, we will add social login later on but low priority. The reason for
this is because we will put this accountgo to some marketplace e.g.
AppDirect and user can subscribe using email and maybe social login. Once
subscribe it iwll automatically setup a company for them.
For authentication/autherization aspnet membership still fine. But we will
extend a permission management to another table.

**Comment by codemvp** [comment]
The way I see it, permissions are hardcoded or added by script on the
back-end. Because this permission are going to be use in the code. Roles
can be created from the UI and permissions can be assign to a role. Roles
are groups of permissions. I'm open for ideas.

**Comment by vivencio2** [comment]
@codemvp okay got you. I will try to implement something and will discuss with you once staged to my fork.
